components/price/price.js

```diff
@@ -0,0 +1,145 @@
+// Imports
+import React from 'react';
+import { getClassNamesWithMods, getDataAttributes } from '../_helpers';
+
+const { PropTypes } = React;
+
+/**
+ * Adds the thousands separator to a given value.
+ * If either the thousands separator is not set or the value doesn't have thousands,
+ * it will return the original value.
+ *
+ * @function addThousandsSeparator
+ * @param {String} value              String to be formatted
+ * @param {String} thousandsSeparator Character to be used to split the thousands unit.
+ * @return {String}                   Either the original value or the formatted value.
+ */
+export function addThousandsSeparator(value, thousandsSeparator) {
+  if (!thousandsSeparator || (value.length <= 3)) {
+    return value;
+  }
+
+  return value
+    .split('')
+    .reverse()
+    .reduce((ret, digit, idx) => ret.concat([((idx > 0) && ((idx % 3) === 0) ? thousandsSeparator : ''), digit]), [])
+    .reverse()
+    .join('');
+}
+
+/**
+ * Ensures that a decimal value has the right decimal precision (no rounding applied).
+ *
+ * @function ensureDecimalPrecision
+ * @param {String} value              String representing the decimal part of a number.
+ * @param {Number} decimalsPrecision  Number of decimals to be allowed.
+ * @return {String}                   The value with a proper precision.
+ */
+export function ensureDecimalPrecision(value, decimalsPrecision = 2) {
+  if (value.length > decimalsPrecision) {
+    return value.substr(0, decimalsPrecision);
+  }
+
+  return `${value}${'0'.repeat(decimalsPrecision - value.length)}`;
+}
+
+/**
+ * Price component
+ */
+function Price({
+  dataAttrs = {},
+  decimalsPrecision,
+  decimalsSeparator,
+  mods = [],
+  showDecimals,
+  symbol,
+  symbolPosition,
+  thousandsSeparator,
+  underlined,
+  value,
+}) {
+  const rootClass = 'ui-price';
+  const className = getClassNamesWithMods(rootClass, mods);
+  const [intValue, decValue] = value.toString().split('.');
+
+  const decimalsMarkup = showDecimals ? (<div className={`${rootClass}__decimals`}>{
+    decimalsSeparator + ensureDecimalPrecision(decValue, decimalsPrecision)
+  }</div>) : null;
+  const underlineMarkup = underlined ? <div className={`${rootClass}__underline`}/> : null;
+
+  return (<div className={className} {...getDataAttributes(dataAttrs)}>
```

Comment:
I would put html to the next line (outside of parentheses), to increase code readability